pkg/controllers/vdb/saverestorepoint_reconciler.go

```diff
@@ -0,0 +1,215 @@
+	if err != nil {
+		// For all other errors, return error
+		s.VRec.Eventf(s.Vdb, corev1.EventTypeWarning, events.CreateArchiveFailed,
+			"Failed to create archive %q", archiveName)
+		return err
+	}
+
+	s.VRec.Eventf(s.Vdb, corev1.EventTypeNormal, events.CreateArchiveSucceeded,
+		"Successfully create archive. It took %s", time.Since(start).Truncate(time.Second))
+	return nil
+}
+
+// runSaveRestorePointVclusterAPI will do the actual execution of saving restore point to
+// an existing archive.
+// This handles logging of necessary events.
+func (s *SaveRestorePoint) runSaveRestorePointVclusterAPI(ctx context.Context,
+	host, sandbox string) error {
+	archiveName := s.Vdb.Spec.RestorePoint.Archive
+	opts := s.genSaveRestorePointOpts(host, archiveName, sandbox)
+	s.VRec.Event(s.Vdb, corev1.EventTypeNormal,
+		events.SaveRestorePointStart, "Starting save restore point")
+	start := time.Now()
+
+	err := s.Dispatcher.SaveRestorePoint(ctx, opts...)
+	if err != nil {
+		s.VRec.Eventf(s.Vdb, corev1.EventTypeWarning, events.SaveRestorePointFailed,
+			"Failed to save restore point to archive: %s", archiveName)
+		return err
+	}
+	end := time.Now()
+	s.VRec.Eventf(s.Vdb, corev1.EventTypeNormal, events.SaveRestorePointSucceeded,
+		"Successfully save restore point to archive: %s. It took %s",
+		archiveName, time.Since(start).Truncate(time.Second))
+	return s.updateVDB(ctx, start, end)
+}
+
+// genCreateArchiveOpts will return the options to use with the create archive apiS
+func (s *SaveRestorePoint) genCreateArchiveOpts(initiatorIP string, archiveName string,
+	numRestorePoint int, sandbox string) []createarchive.Option {
+	opts := []createarchive.Option{
+		createarchive.WithInitiator(initiatorIP),
+		createarchive.WithArchiveName(archiveName),
+		createarchive.WithNumRestorePoints(numRestorePoint),
+		createarchive.WithSandbox(sandbox),
+	}
+	return opts
+}
+
+// genSaveRestorePointOpts will return the options to use with the save restore point api
+func (s *SaveRestorePoint) genSaveRestorePointOpts(initiatorIP string, archiveName string,
+	sandbox string) []saverestorepoint.Option {
+	opts := []saverestorepoint.Option{
+		saverestorepoint.WithInitiator(initiatorIP),
+		saverestorepoint.WithArchiveName(archiveName),
+		saverestorepoint.WithSandbox(sandbox),
+	}
+	return opts
+}
+
+// updateVDB clears the condition status and add the restore point details
+// in the status.
+func (s *SaveRestorePoint) updateVDB(ctx context.Context, start, end time.Time) error {
+	refreshStatusInPlace := func(vdb *vapi.VerticaDB) error {
+		meta.SetStatusCondition(&vdb.Status.Conditions, *vapi.MakeCondition(vapi.SaveRestorePointNeeded,
+			metav1.ConditionFalse, "Completed"))
+		if vdb.Status.RestorePoint == nil {
+			vdb.Status.RestorePoint = new(vapi.RestorePointInfo)
+		}
+		vdb.Status.RestorePoint.Archive = s.Vdb.Spec.RestorePoint.Archive
+		vdb.Status.RestorePoint.StartTimestamp = start.Format("2006-01-02 15:04:05.000000000")
```

Comment:
Create a separate Jira to fetch the restore point's ID or index, then put the ID or index in the status rather than start-time and end-time.

Reply:
Ok, but Users can use the start and end timestamp to get query the restorepoints with that CR(vrpq) and get the id/index info. This is good enough for this release.

Reply:
Yes, that will be a follow-up Jira for 25.1 or 25.2.